<launch>
    <rosparam file="$(find suitbot_ros)/config/suitbot.yaml" />

    <arg name="use_rviz"  default="true"/> 

    <arg name="manual_control"        default="true"/> 
    <param name="manual_control"      type="bool"   value="$(arg manual_control)"/>

    <arg name="debug_odometry"        default="false"/> 
    <param name="debug_odometry"      type="bool"   value="$(arg debug_odometry)"/>
    <arg name="debug_time"            default="5.0"/> 
    <param name="debug_time"          type="double"   value="$(arg debug_time)"/>

        <arg name="debug_linear"          default="0.5"/> 
        <param name="debug_linear"        type="double"   value="$(arg debug_linear)"/>
        <arg name="debug_angular"         default="0.0"/> 
        <param name="debug_angular"              type="double"   value="$(arg debug_angular)"/>

        <arg name="init_x"                default="0.0"/> 
        <param name="init_x"              type="double"   value="$(arg init_x)"/>
        <arg name="init_y"                default="0.0"/> 
        <param name="init_y"              type="double"   value="$(arg init_y)"/>
        <arg name="init_theta"            default="0.0"/> 
        <param name="init_theta"          type="double"   value="$(arg init_theta)"/>

    <arg name="use_audio"             default="false"/> 
    <param name="use_audio"           type="bool"   value="$(arg use_audio)"/>
    <arg name="course_idx"            default="0"/> 
    <param name="course_idx"          type="int"   value="$(arg course_idx)"/>
    

    <arg name="use_serial"            default="false"/> 
    <param name="use_serial"           type="bool"   value="$(arg use_serial)"/>
    <param name="map_file" value="$(find suitbot_ros)/data/wean_map_grid_edited.jpg"/>
    <param name="pcd_file" value="$(find suitbot_ros)/data/wean.pcd"/>
    <param name="model_path" value="$(find suitbot_ros)/scripts/audio/vosk-model-small-en-us-0.15"/>

    
    <node name="lidar_driver" pkg="suitbot_ros" type="lidar_driver.py" output="screen"/>
    <node name="lidar_parse" pkg="suitbot_ros" type="lidar_parse" output="screen"/>
    <node name="tracking_simulator" pkg="suitbot_ros" type="tracking_simulator.py" output="screen"/>
    
    <group if="$(arg use_rviz)">
        <!-- stuff that will only be evaluated if foo is true -->
        <node name="rvizvisualisation" pkg="rviz" type="rviz" output="log" args="-d $(find suitbot_ros)/rviz/local_map_gen.rviz" />
    </group>
</launch>
